modulus/utils/capture.py

```diff
@@ -32,18 +32,20 @@ class _StaticCapture(object):
     should be used instead for training and evaluation functions.
     """
 
-    # Grad scalar singleton use for checkpointing
-    # This limits the number of staticcapture AMP training instances to just one per program
-    scaler_dict = None
-    scaler_singleton = None
+    # Grad scaler and checkpoint singleton use for checkpoint saving and loading
+    # Since an instance of Static capture does not exist for checkpoint functions
+    # one must use class functions to access state dicts
+    _amp_scalers = {}
+    _amp_scaler_checkpoints = {}
```

Comment:
If the intent is for this class to be a singleton, shouldn't this be enforced through either a singleton or Borg pattern?

Reply:
Yeah, doc was not correct, was more so a class function. But switched to singleton approach